src/imputation/pg_conversion.py

```
@@ -0,0 +1,157 @@
+    )
+
+    PgLogger.info("Product group nulls successfully mapped from SIC.")
+
+    return df
+
+
+def pg_to_pg_mapper(
+    df: pd.DataFrame,
+    mapper: pd.DataFrame,
+    pg_column: str = "201",
+    from_col: str = "pg_numeric",
+    to_col: str = "pg_alpha",
+):
+    """Map from PG numeric to PG alpha-numeric and create a new column.
+
+    The product group column (default: column 201) coped to a new column, "pg_numeric",
+    and then is updated from numeric to alpha-numeric using a mapping.
+
+    Example initial dataframe:
+        reference | 201     
+    ----------------------
+        1         | 53    
+        2         | 43     
+        3         | 33    
+
+    returned dataframe:
+        reference | 201     | pg_numeric
+    ------------------------------------
+        1         | AA      | 33
+        2         | B       | 43
+        3         | E       | 53
+
+
+    Args:
+        df (pd.DataFrame): The dataframe requiring mapping
+        mapper (pd.DataFrame): the PG numeric to alpha-numeric mapper
+        pg_column (str, optional): The column we want to convert (default 201).
+        from_col (str, optional): The column in the mapper that is used to map from.
+        to_col (str, optional): The column in the mapper that is used to map to.
+
+    Returns:
+        pd.DataFrame: A dataframe with all target column values mapped
+    """
+
+    df = df.copy()
+
+    # Copy the numeric PG column to a new column
+    df["pg_numeric"] = df[pg_column].copy()
+
+    # Create a mapping dictionary from the 2 columns
+    map_dict = dict(zip(mapper[from_col], mapper[to_col]))
+
+    # Flag all PGs that don't have a corresponding map value
+    mapless_errors = []
+    for key, value in map_dict.items():
+        if str(value) == "nan":
+            mapless_errors.append(key)
+
+    if mapless_errors:
+        PgLogger.error(
+            f"Mapping doesnt exist for the following product groups: {mapless_errors}"
+        )
```

As above, just wanted to check whether this should raise an exception or not.